### +++ Visualización de tabla completa +++

In [65]:
import pandas as pd

datos = pd.read_csv('avocado.csv',delimiter=',',decimal=".",index_col=1)
print(datos.shape)
datos

(18249, 13)


Unnamed: 0  AveragePrice  Total Volume     4046       4225  \
Date                                                                     
2015-12-27           0          1.33      64236.62  1036.74   54454.85   
2015-12-20           1          1.35      54876.98   674.28   44638.81   
2015-12-13           2          0.93     118220.22   794.70  109149.67   
2015-12-06           3          1.08      78992.15  1132.00   71976.41   
2015-11-29           4          1.28      51039.60   941.48   43838.39   
...                ...           ...           ...      ...        ...   
2018-02-04           7          1.63      17074.83  2046.96    1529.20   
2018-01-28           8          1.71      13888.04  1191.70    3431.50   
2018-01-21           9          1.87      13766.76  1191.92    2452.79   
2018-01-14          10          1.93      16205.22  1527.63    2981.04   
2018-01-07          11          1.62      17489.58  2894.77    2356.13   

              4770  Total Bags  Small Bags  Large Bags  XLarge Bags  \
Date                                                                  
2015-12-27   48.16     8696.87     8603.62       93.25          0.0   
2015-12-20   58.33     9505.56     9408.07       97.49          0.0   
2015-12-13  130.50     8145.35     8042.21      103.14          0.0   
2015-12-06   72.58     5811.16     5677.40      133.76          0.0   
2015-11-29   75.78     6183.95     5986.26      197.69          0.0   
...            ...         ...         ...         ...          ...   
2018-02-04    0.00    13498.67    13066.82      431.85          0.0   
2018-01-28    0.00     9264.84     8940.04      324.80          0.0   
2018-01-21  727.94     9394.11     9351.80       42.31          0.0   
2018-01-14  727.01    10969.54    10919.54       50.00          0.0   
2018-01-07  224.53    12014.15    11988.14       26.01          0.0   

                    type  year            region  
Date                                              
2015-12-27  conventional  2015            Albany  
2015-12-20  conventional  2015            Albany  
2015-12-13  conventional  2015            Albany  
2015-12-06  conventional  2015            Albany  
2015-11-29  conventional  2015            Albany  
...                  ...   ...               ...  
2018-02-04       organic  2018  WestTexNewMexico  
2018-01-28       organic  2018  WestTexNewMexico  
2018-01-21       organic  2018  WestTexNewMexico  
2018-01-14       organic  2018  WestTexNewMexico  
2018-01-07       organic  2018  WestTexNewMexico  

[18249 rows x 13 columns]

### +++ Visualización de las primeras líneas de tabla (suando función head) +++

In [75]:
import pandas as pd 

datos = pd.read_csv("avocado.csv",delimiter=',',decimal=".",index_col=1)
#Para cambiar el orden de las columnas, modificamos el valor de: index_col 

datos.head(20)

Unnamed: 0  AveragePrice  Total Volume     4046       4225  \
Date                                                                     
2015-12-27           0          1.33      64236.62  1036.74   54454.85   
2015-12-20           1          1.35      54876.98   674.28   44638.81   
2015-12-13           2          0.93     118220.22   794.70  109149.67   
2015-12-06           3          1.08      78992.15  1132.00   71976.41   
2015-11-29           4          1.28      51039.60   941.48   43838.39   
2015-11-22           5          1.26      55979.78  1184.27   48067.99   
2015-11-15           6          0.99      83453.76  1368.92   73672.72   
2015-11-08           7          0.98     109428.33   703.75  101815.36   
2015-11-01           8          1.02      99811.42  1022.15   87315.57   
2015-10-25           9          1.07      74338.76   842.40   64757.44   
2015-10-18          10          1.12      84843.44   924.86   75595.85   
2015-10-11          11          1.28      64489.17  1582.03   52677.92   
2015-10-04          12          1.31      61007.10  2268.32   49880.67   
2015-09-27          13          0.99     106803.39  1204.88   99409.21   
2015-09-20          14          1.33      69759.01  1028.03   59313.12   
2015-09-13          15          1.28      76111.27   985.73   65696.86   
2015-09-06          16          1.11      99172.96   879.45   90062.62   
2015-08-30          17          1.07     105693.84   689.01   94362.67   
2015-08-23          18          1.34      79992.09   733.16   67933.79   
2015-08-16          19          1.33      80043.78   539.65   68666.01   

              4770  Total Bags  Small Bags  Large Bags  XLarge Bags  \
Date                                                                  
2015-12-27   48.16     8696.87     8603.62       93.25          0.0   
2015-12-20   58.33     9505.56     9408.07       97.49          0.0   
2015-12-13  130.50     8145.35     8042.21      103.14          0.0   
2015-12-06   72.58     5811.16     5677.40      133.76          0.0   
2015-11-29   75.78     6183.95     5986.26      197.69          0.0   
2015-11-22   43.61     6683.91     6556.47      127.44          0.0   
2015-11-15   93.26     8318.86     8196.81      122.05          0.0   
2015-11-08   80.00     6829.22     6266.85      562.37          0.0   
2015-11-01   85.34    11388.36    11104.53      283.83          0.0   
2015-10-25  113.00     8625.92     8061.47      564.45          0.0   
2015-10-18  117.07     8205.66     7877.86      327.80          0.0   
2015-10-11  105.32    10123.90     9866.27      257.63          0.0   
2015-10-04  101.36     8756.75     8379.98      376.77          0.0   
2015-09-27  154.84     6034.46     5888.87      145.59          0.0   
2015-09-20  150.50     9267.36     8489.10      778.26          0.0   
2015-09-13  142.00     9286.68     8665.19      621.49          0.0   
2015-09-06  240.79     7990.10     7762.87      227.23          0.0   
2015-08-30  335.43    10306.73    10218.93       87.80          0.0   
2015-08-23  444.78    10880.36    10745.79      134.57          0.0   
2015-08-16  394.90    10443.22    10297.68      145.54          0.0   

                    type  year  region  
Date                                    
2015-12-27  conventional  2015  Albany  
2015-12-20  conventional  2015  Albany  
2015-12-13  conventional  2015  Albany  
2015-12-06  conventional  2015  Albany  
2015-11-29  conventional  2015  Albany  
2015-11-22  conventional  2015  Albany  
2015-11-15  conventional  2015  Albany  
2015-11-08  conventional  2015  Albany  
2015-11-01  conventional  2015  Albany  
2015-10-25  conventional  2015  Albany  
2015-10-18  conventional  2015  Albany  
2015-10-11  conventional  2015  Albany  
2015-10-04  conventional  2015  Albany  
2015-09-27  conventional  2015  Albany  
2015-09-20  conventional  2015  Albany  
2015-09-13  conventional  2015  Albany  
2015-09-06  conventional  2015  Albany  
2015-08-30  conventional  2015  Albany  


Nota: Para poder responder a la 2da pregunta que se ecuentra al final de este código, se puso como argumento "20"

### +++ Combinaciones de columnas y datos +++ 

In [67]:
datos.iloc[[1,2,3],[4,5,6]]

4225    4770  Total Bags
Date                                     
2015-12-20   44638.81   58.33     9505.56
2015-12-13  109149.67  130.50     8145.35
2015-12-06   71976.41   72.58     5811.16

### +++ Usando la función "describe" +++

In [68]:
print(datos.describe())

         Unnamed: 0  AveragePrice  Total Volume          4046          4225  \
count  18249.000000  18249.000000  1.824900e+04  1.824900e+04  1.824900e+04   
mean      24.232232      1.405978  8.506440e+05  2.930084e+05  2.951546e+05   
std       15.481045      0.402677  3.453545e+06  1.264989e+06  1.204120e+06   
min        0.000000      0.440000  8.456000e+01  0.000000e+00  0.000000e+00   
25%       10.000000      1.100000  1.083858e+04  8.540700e+02  3.008780e+03   
50%       24.000000      1.370000  1.073768e+05  8.645300e+03  2.906102e+04   
75%       38.000000      1.660000  4.329623e+05  1.110202e+05  1.502069e+05   
max       52.000000      3.250000  6.250565e+07  2.274362e+07  2.047057e+07   

               4770    Total Bags    Small Bags    Large Bags    XLarge Bags  \
count  1.824900e+04  1.824900e+04  1.824900e+04  1.824900e+04   18249.000000   
mean   2.283974e+04  2.396392e+05  1.821947e+05  5.433809e+04    3106.426507   
std    1.074641e+05  9.862424e+05  7.461785e+05 

## +++ Preguntas +++

###### 1. ¿Cuántos objetos hay?

Hay: 18 249 objetos

###### 2. ¿Cuál es el valor de la variable 2 del objeto 15?

El valor es: 1.33

###### 3. ¿Cuál es el mínimo y máximo de la variable 3?

In [45]:
print("================Máximos===============")
print(datos.max(numeric_only=True))
print("\n================Mínimos===============")
print(datos.min(numeric_only=True))

================Máximos===============
Unnamed: 0            52.00
AveragePrice           3.25
Total Volume    62505646.52
4046            22743616.17
4225            20470572.61
4770             2546439.11
Total Bags      19373134.37
Small Bags      13384586.80
Large Bags       5719096.61
XLarge Bags       551693.65
year                2018.00
dtype: float64

================Mínimos===============
Unnamed: 0         0.00
AveragePrice       0.44
Total Volume      84.56
4046               0.00
4225               0.00
4770               0.00
Total Bags         0.00
Small Bags         0.00
Large Bags         0.00
XLarge Bags        0.00
year            2015.00
dtype: float64


**********************************
Máximo = 62505646.52
**********************************
Mínimo = 84.56